Sources/AWSSignerV4/signer.swift

```diff
@@ -19,23 +20,23 @@ public struct AWSSigner {
     public let name: String
     /// AWS region you are working in
     public let region: String
-    
-    static let hashedEmptyBody = AWSSigner.hexEncoded(sha256([UInt8]()))
-    
+
+    static let hashedEmptyBody = SHA256.hash(data: [UInt8]()).hexDigest()
+
     /// Initialise the Signer class with AWS credentials
     public init(credentials: Credential, name: String, region: String) {
         self.credentials = credentials
         self.name = name
         self.region = region
     }
-    
+
     /// Enum for holding your body data
     public enum BodyData {
```

Comment:
While we are at it, I would love to see this removed and be replaced with a protocol. I even think that 
```
<Bytes: Collection> where Bytes.Element == UInt8
```

should be enough.

Reply:
I put down an example of the new method signature. I think this would make the signer code way better, and would lead to a signer, that maybe is only depended on `Crypto`. And we could make this dependency even

```swift
@_implementationOnly import Crypto
```

Reply:
> While we are at it, I would love to see this removed and be replaced with a protocol. I even think that
> 
> ```
> <Bytes: Collection> where Bytes.Element == UInt8
> ```
> 
> should be enough.

Isn't that simple. As the body is optional there are situations where it cannot ascertain what object `Bytes` is. Would need to have separate functions, one that takes a non optional buffer and one that does not. It is doable just more work.

Reply:
I've add another PR for this so you can see this done as separate functions #195

Reply:
Thought I tested it, obviously not. It fails for `Data` as `withContiguousStorageIfAvailable` doesn't work for `Data`. There is no other way to get a raw pointer with `Collection`. Don't think we can generalize this for `Collections`